typescript/sdk/src/ism/EvmIsmModule.ts

```diff
@@ -86,4 +186,610 @@ export class EvmIsmModule extends HyperlaneModule<
       config,
     });
   }
+
+  protected async updateRoutingIsm(params: {
+    destination: ChainName;
+    config: RoutingIsmConfig;
+    origin?: ChainName;
+    mailbox?: Address;
+    existingIsmAddress: Address;
+    logger: Logger;
+  }): Promise<IRoutingIsm> {
+    const { destination, config, mailbox, existingIsmAddress, logger } = params;
+
+    // 1. Filter for known domains
+    const { domains, safeConfigDomains } = EvmIsmModule.filterRoutingIsmDomains(
+      { config, multiProvider: this.multiProvider },
+    );
+    config.domains = domains;
+
+    // 2. Compute the delta
+    const delta: RoutingIsmDelta = await routingModuleDelta(
+      destination,
+      existingIsmAddress,
+      config,
+      this.multiProvider,
+      this.factories,
+      mailbox,
+    );
+
+    // Step 3: Get owner and signer
+    const signer = this.multiProvider.getSigner(destination);
+    const provider = this.multiProvider.getProvider(destination);
+    const owner = await DomainRoutingIsm__factory.connect(
+      existingIsmAddress,
+      provider,
+    ).owner();
+    const isOwner = eqAddress(await signer.getAddress(), owner);
+
+    if (existingIsmAddress && isOwner && !delta.mailbox) {
+      // 4. If possible, reconfigure existing ISM
+      return await this.reconfigureExistingRoutingIsm({
+        destination,
+        delta,
+        existingIsmAddress,
+        config,
+        mailbox,
+        logger,
+      });
+    } else {
+      // 5. Else, deploy a new set of routing ISM submodules
+      const submoduleAddresses = await this.deployRoutingIsmSubmodules({
+        destination,
+        config,
+        mailbox,
+      });
+
+      const owner = await resolveOrDeployAccountOwner(
+        this.multiProvider,
+        destination,
+        config.owner,
+      );
+
+      if (config.type === IsmType.FALLBACK_ROUTING) {
+        if (!mailbox) {
+          throw new Error(
+            'Mailbox address is required for deploying fallback routing ISM',
+          );
+        }
+
+        // 6. If fallback routing ISM, reinitialise in-place with new submodules
+        return await this.initializeFallbackRoutingIsm({
+          destination,
+          owner,
+          safeConfigDomains,
+          submoduleAddresses,
+          existingIsmAddress,
+          signer,
+          logger,
+        });
+      } else {
+        // 7. Else, deploy a new domain routing ISM
+        return await EvmIsmModule.deployDomainRoutingIsm({
+          destination,
+          owner,
+          safeConfigDomains,
+          submoduleAddresses,
+          overrides: this.multiProvider.getTransactionOverrides(destination),
+          multiProvider: this.multiProvider,
+          factories: this.args.addresses,
+        });
+      }
+    }
+  }
+
+  protected async reconfigureExistingRoutingIsm(params: {
```

Comment:
how is this different from update?

Reply:
I didn't want each function to be super long so we have update() -> if condition: updateRoutingIsm() -> if condition: reconfigureExistingRoutingIsm() 

strictly speaking the logic in `updateRoutingIsm` could go into `update` and `reconfigureExistingRoutingIsm` should be renamed to `updateRoutingIsm` - but then this adds